z390test/src/main/groovy/org/z390/test/z390Test.groovy

```diff
@@ -0,0 +1,267 @@
+package org.z390.test
+
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+
+class z390Test {
+
+    var project_root = pathJoin('..')
+    String stdout = ""
+    String stderr = ""
+    var env = [:]
+    var fileData = [:]
+    var message = "z390Test - Test framework for z390 project"
+    int cmdTimeMs = 100000
+    boolean printOutput = false
+
+    File tempDir = null
+
+    static void main(String[] args) {
+        println new z390Test().message
+    }
+
+    z390Test() {
+        /**
+         * class constructor
+         */
+        if (System.getenv('Z390_PROJECT_ROOT')) {
+            this.project_root = new File(System.getenv('Z390_PROJECT_ROOT'))
+        }
+        if (System.getenv('Z390_PRINT_OUTPUT')) {
+            printOutput = true
+        }
+    }
+    @BeforeEach
+    void setUp() {
+        this.stdout = ""
+        this.stderr = ""
+        this.fileData = [:]
+    }
+    @AfterEach
+    void cleanUp() {
+        if (this.tempDir)
+            this.tempDir.deleteDir()
```

Comment:
Is the if statement merely checking that tempDir is not null?  If "yes", then the directory will be deleted but variable tempDir will still be not null, won't it? Might this cause a problem in the following createTempFile() method? Just asking.

Reply:
The first time a temp file is needed in a test class using the createTempFile method, a directory name assigned to the tempFile class variable and the directory created.
```groovy
if (!tempDir) {
  this.tempDir = File.createTempDir()
  this.tempDir.deleteOnExit()
}
```
At completion of the first test that created a temp file, the directory is deleted in the cleanup method, but the tempDir is still pointing to the dir name used.
```groovy
    @AfterEach
    void cleanUp() {
        if (this.tempDir)
            this.tempDir.deleteDir()
    }
```
tempDir now is not null so a new directory or name is not created on second and subsequent calls to create temp files.. But the method that creates the file will use the directory name and presumably create the directory if it does not exist.
```groovy
        new File(fullFileName).with {
            createNewFile()
            write(fileContents)
        }
```
This is what I guess is happening, but would need to spend some time to prove this is the case..
Its a bit sloppy and could probably be done a few different ways but it does seem to be working so
if it ain't broke, don't fix it...

Reply:
@jyganci -- let me know your thoughts on this and if I need to make some more changes

Reply:
My only concern is the use of the tempDir variable after the directory is deleted.
I am assuming that you've tested the reuse of tempDir after a deletion of the
directory. If so, then I'll agree that there is not an issue here.